### PR TITLE
Add CI/CD and semantic release

### PR DESCRIPTION
**Pull request by rmi22186**
## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
We did not have semantic release on this kit.  Since we don't update these kits too often, I wanted to implement this now.  Most of this is pulled and adapted from GA4.  For reference below:
* release.config.js - https://github.com/mparticle-integrations/mparticle-javascript-integration-google-analytics-4/blob/master/packages/GA4Client/release.config.js
* release.yml - https://github.com/mparticle-integrations/mparticle-javascript-integration-google-analytics-4/blob/master/.github/workflows/release.yml

 ## Testing Plan
 - [ ] Was this tested locally? If not, explain why.
It is not possible to test a github action prior to release. However, if there are any failures, we can debug once this file is in `master`.

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-5944

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [10.x, 12.x]
+        node-version: [18.x, 20.x]
 
     steps:
     - uses: actions/checkout@v2
```
Comment (alexs-mparticle) [inline .github/workflows/pull-request.yml]: I believe you should update this file as part of your Karma test update. However, please make sure to update all the references to `v2` actions to `v3` as this version includes a more current node version.

File: release.config.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        [
+            '@semantic-release/release-notes-generator',
+            {
+                preset: 'angular',
+            },
+        ],
+        [
+            '@semantic-release/changelog',
+            {
+                changelogFile: 'CHANGELOG.md',
```
Comment (alexs-mparticle) [inline release.config.js:34]: I may be wrong here, but do you need to include `CHANGELOG.md` manually, or will this be generated for you the first time?

Comment (alexs-mparticle) [reply release.config.js:34]: Also, can you please include `CONTRIBUTING.md`?

**Review by alexs-mparticle — CHANGES_REQUESTED** [review]

File: release.config.js (hunk 2 above)
Comment (rmi22186) [reply release.config.js:34]: Good call about CHANGELOG.md.    I actually don't know either, so I'll add one now.

File: package.json [hunk 3]
```diff
@@ -11,18 +11,30 @@
   "repository": "https://github.com/mparticle-integrations/mparticle-javascript-integration-criteo",
   "scripts": {
     "build": "rollup --config rollup.config.js",
-    "watch": "rollup --config rollup.config.js -w"
+    "watch": "rollup --config rollup.config.js -w",
+    "build:test": "ENVIRONMENT=production rollup --config rollup.test.config.js",
```
Comment (alexs-mparticle) [inline package.json]:
```suggestion
    "build:test": "rollup --config rollup.test.config.js",
```

**Review by alexs-mparticle — CHANGES_REQUESTED** [review]

**Review by alexs-mparticle — APPROVED** [review]

**Pull request merged**